discovery/nomad/nomad.go

```
@@ -0,0 +1,181 @@
+	prometheus.MustRegister(failuresCount)
+}
+
+// SDConfig is the configuration for nomad based service discovery.
+type SDConfig struct {
+	HTTPClientConfig config.HTTPClientConfig `yaml:",inline"`
+	Namespace        string                  `yaml:"namespace"`
+	RefreshInterval  model.Duration          `yaml:"refresh_interval"`
+	Region           string                  `yaml:"region"`
+	Server           string                  `yaml:"server"`
+}
+
+// Name returns the name of the Config.
+func (*SDConfig) Name() string { return "nomad" }
+
+// NewDiscoverer returns a Discoverer for the Config.
+func (c *SDConfig) NewDiscoverer(opts discovery.DiscovererOptions) (discovery.Discoverer, error) {
+	return NewDiscovery(c, opts.Logger)
+}
+
+// SetDirectory joins any relative file paths with dir.
+func (c *SDConfig) SetDirectory(dir string) {
+	c.HTTPClientConfig.SetDirectory(dir)
+}
+
+// UnmarshalYAML implements the yaml.Unmarshaler interface.
+func (c *SDConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	*c = DefaultSDConfig
+	type plain SDConfig
+	err := unmarshal((*plain)(c))
+	if err != nil {
+		return err
+	}
+	return c.HTTPClientConfig.Validate()
+}
+
+// Discovery periodically performs nomad requests. It implements
+// the Discoverer interface.
+type Discovery struct {
+	*refresh.Discovery
+	client          *nomad.Client
+	namespace       string
+	refreshInterval time.Duration
+	region          string
+	server          string
+}
+
+// NewDiscovery returns a new Discovery which periodically refreshes its targets.
+func NewDiscovery(conf *SDConfig, logger log.Logger) (*Discovery, error) {
+	d := &Discovery{
+		namespace:       conf.Namespace,
+		refreshInterval: time.Duration(conf.RefreshInterval),
+		region:          conf.Region,
+		server:          conf.Server,
+	}
+
+	config := nomad.Config{
+		Address:   conf.Server,
+		Namespace: conf.Namespace,
+		Region:    conf.Region,
+	}
+
+	client, _ := nomad.NewClient(&config)
```

Please do not discard the error here

---

fixed in 37964023